lua/telescope/previewers.lua

```diff
@@ -338,6 +338,32 @@ previewers.vim_buffer = defaulter(function(_)
   }
 end, {})
 
+previewers.git_commit_diff = defaulter(function(_)
+  return previewers.new_termopen_previewer {
+    get_command = function(entry)
+      local sha = entry.value
+      return { 'git', '--no-pager', 'diff', sha .. '^!' }
+    end
+  }
+end, {})
+
+previewers.git_branch_log = defaulter(function(_)
+  return previewers.new_termopen_previewer {
+    get_command = function(entry)
+      return { 'git', 'log', '--graph',
+               '--pretty=format:%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr)%Creset',
+               '--abbrev-commit', '--date=relative', entry.value }
+    end
+  }
+end, {})
```

Comment:
Does it make sense to show a git log as branch preview. I tried to make it "pretty" with graph and colors.

Reply:
git log sounds like a fun way to do it!